runtime/src/iree/hal/cts/cts_test_base.h

```diff
@@ -276,6 +277,20 @@ class CTSTestBase : public BaseType, public CTSTestResources {
     IREE_EXPECT_OK(iree_hal_semaphore_query(semaphore, &value));
     EXPECT_EQ(expected_value, value);
   }
+
+  void CheckStatusEqual(iree_status_t a, iree_status_t b) {
```

Comment:
This looks like the sort of helper that belongs in `status.h` or `status_cc.h`, if it is needed at all.

The test code right now assumes that the underlying code will return the error as-is, with no additional status chaining (`iree_status_annotate`). Do we want the tests to be this specific, or should they just have that the result code and _possibly_ some of the message makes it through?

```c
  iree_status_t status =
      iree_make_status(IREE_STATUS_CANCELLED, "FailThenWait test.");
  iree_hal_semaphore_fail(semaphore, iree_status_clone(status));

  iree_status_t wait_status = iree_hal_semaphore_wait(
      semaphore, 1, iree_make_deadline(IREE_TIME_INFINITE_FUTURE));
  EXPECT_EQ(iree_status_code(wait_status), IREE_STATUS_ABORTED);
  uint64_t value = 1234;
  iree_status_t query_status = iree_hal_semaphore_query(semaphore, &value);
  EXPECT_EQ(value, IREE_HAL_SEMAPHORE_FAILURE_VALUE);
  CheckStatusEqual(status, query_status);
```

Reply:
I relaxed the test to allow of message containing, not a exact match.